catalog/core/catalog-core-api-impl/src/test/groovy/ddf/catalog/util/impl/ResultIterableSpec.groovy

```diff
@@ -188,38 +192,105 @@ class QueryResultPaginatorSpec extends Specification {
         Query queryMock = createQueryMock(1, 0)
         QueryRequest queryRequestMock = createQueryRequestMock(queryMock)
 
-        def resultIterator = new ResultIterable(catalogFramework, queryRequestMock).iterator()
+        def resultIterator = resultIterable(catalogFramework, queryRequestMock).iterator()
         def results = []
 
         when:
         1.upto(resultSize + 1, { results << resultIterator.next() })
 
         then:
-        1.upto(resultSize + 1, { results.get(it - 1).distanceInMeters == (Double) it })
+        1.upto(resultSize + 1, { assert results.get(it - 1).distanceInMeters == (Double) it })
+    }
+
+    def "Properly pages when default page size is used with no maxResults"() {
+        setup:
+        def totalResults = 25
+        def resultSize = 20
+
+        3 * catalogFramework.query(_ as QueryRequest) >> {
+            QueryRequest queryRequest -> buildQueryResponse(queryRequest, resultSize, totalResults)
+        } >> {
+            QueryRequest queryRequest -> buildQueryResponse(queryRequest, 5, totalResults)
+        } >> {
+            QueryRequest queryRequest -> buildQueryResponse(queryRequest, 0, totalResults)
+        }
+
+        Query queryMock = createQueryMock(1, 0)
+        QueryRequest queryRequestMock = createQueryRequestMock(queryMock)
+
+        ResultIterable resultIterable = resultIterable(catalogFramework, queryRequestMock)
+
+        when:
+        def results = resultIterable.stream().collect(Collectors.toList())
+
+
+        then:
+        results.size() == 25
+    }
+
+    @Unroll
+    def "Properly pages when default page size is used with maxResults"() {
+        setup:
+        def actualResults = (1..25).collect { new ResultImpl() }
+        def totalResults = actualResults.size()
+
+        // pageSize = 20
+        expectedQueriesMade * catalogFramework.query(_ as QueryRequest) >>
+                { QueryRequest queryRequest -> buildQueryResponse(actualResults[0..19]) } >>
+                { QueryRequest queryRequest -> buildQueryResponse(actualResults[20..24]) } >>
+                { QueryRequest queryRequest -> buildQueryResponse([]) }
+
+
+
+        Query queryMock = createQueryMock(1, 0)
+        QueryRequest queryRequestMock = createQueryRequestMock(queryMock)
+
+        ResultIterable resultIterable = ResultIterable.resultIterable(catalogFramework, queryRequestMock, expectedResultCount)
+
+        when:
+        def queryResults = resultIterable.stream()
+                .collect(Collectors.toList())
+
+        then:
+        def resultCount = Math.min(expectedResultCount, totalResults)
+        // Either expect the maxResults it was capped to, or the actual result count if it was less
+        queryResults.size() == resultCount
+        queryResults == actualResults.subList(0, resultCount)
```

Comment:
❓ Is this equivalent to comparing the `distanceInMeters` like we do in the other tests? If so, maybe we should use this approach instead of the one I've used for the other tests. I like this one better :-)